nservicebus/upgrades/release-policy.md

```diff
@@ -49,7 +49,15 @@ It is strongly recommended to upgrade frequently enough to stay on a supported v
 
 ### Deprecation
 
-Public API methods and properties are deprecated using the [ObsoleteAttribute messages](https://msdn.microsoft.com/en-us/library/system.obsoleteattribute.aspx). Continued usage of a deprecated method or property will result either in a compiler warning or error. A warning message indicates the version in which the deprecation will start to generate a compiler error (usually the next major version). The message will also indicate the version in which the deprecated method or property will be removed. Deprecations in minor releases will always generate a compiler warning and not an error.
+Public APIs are deprecated using [ObsoleteAttribute](https://msdn.microsoft.com/en-us/library/system.obsoleteattribute.aspx) messages. Continued usage of a deprecated API will result in either a compiler warning or error. A warning message indicates the version in which the deprecation will start to generate a compiler error and in which version the API will be removed. For example:
+
+> warning CS0618: 'CorrelationContextExtensions.SetCorrelationId(SendOptions, string)' is obsolete: 'Setting a custom correlation ID is discouraged and will be removed in the next major version. Will be treated as an error from version 7.0.0. Will be removed in version 8.0.0.'
```

Comment:
@Particular/nservicebus-maintainers - can you think of a better warning one that we introduced in Version 5 that was removed in Version 6? Something that our users are familiar with?